File: 0945-minimum-increment-to-make-array-unique/0945-minimum-increment-to-make-array-unique.cpp
class Solution {
public:
    int minIncrementForUnique(vector<int>& arr) {
        int n = arr.size();
        int ans = 0;
        sort(arr.begin(), arr.end());
        for(int i=1; i<n; i++){
            if(arr[i]<=arr[i-1]){
                ans += arr[i-1] + 1 - arr[i];
                arr[i] = arr[i-1] + 1;
            }
        }
        return ans;
    }
};

File: 0945-minimum-increment-to-make-array-unique/0945-minimum-increment-to-make-array-unique.go
func minIncrementForUnique(arr []int) int {
    ans := 0;
    sort.Ints(arr);
    for i := 1; i<len(arr); i++ {
        if(arr[i]<=arr[i-1]){
            ans += arr[i-1] + 1 - arr[i];
            arr[i] = arr[i-1]+1;
        }
    }
    return ans;
}